freqtrade/tests/test_freqtradebot.py

```diff
@@ -706,14 +727,14 @@ def test_process_operational_exception(default_conf, ticker, markets, mocker) ->
         markets=PropertyMock(return_value=markets),
         buy=MagicMock(side_effect=OperationalException)
     )
-    freqtrade = FreqtradeBot(default_conf)
-    patch_get_signal(freqtrade)
+    worker = get_patched_worker(mocker, default_conf)
```

Comment:
The problem here is that you use a patched freqtradebot instead of a real one.
The "patched" bot has persistence mocked away - so db does not work but is required for this test.

same problem is with the other failing test.
I've fixed that for you - but want you to know what the problem was.

Note that there are still a few flake / mypy errors around. Also, I did look at these tests only. I will need to find more time to do a full review - thinking through the logic ... which I didn't have time for yet, but i HOPE to find over the weekend.

Reply:
Thanks alot.

I also do not have much spare time in the meantime, but will try to describe before weekend in details some of the issues I do not like in the current implementation (current I mean this one, after refactoring). I'd definetely appreciate reviewing the implementation as a whole, there are a few doubtful moments...

Reply:
I know about mypy and flake8 points, I intentionally have not looked at them yet.

By the way, I also think that the tests that are now related to the Worker class should also be moved to a separate testing .py. I intentionally left it for the future because it much easier to review the changes before such wide movement of the code...